### ISSUE TITLE
Rename `torch.nn.MSELoss` to `torch.nn.SELoss`

### ISSUE THREAD

**Issue by goelakash** [pytorch/pytorch#13551]
From the documentation [here](https://pytorch.org/docs/stable/nn.html), it seems that the loss is mainly the element-wise squared differences between two vectors.
The reduction function should be able to handle special cases where `mean`, `sum` or no-reduction is required.

Does this argument make sense?

### Please note that this issue tracker is not a help form and this issue will be closed.

We have a set of [listed resources available on the website](https://pytorch.org/resources). Our primary means of support is our discussion forum:

- [Discussion Forum](https://discuss.pytorch.org/)

**Comment by ragulpr** [comment]
What would `SELoss`-acronym stand for? Mean Squared Error is well known in stat/ml-community so it makes sense. Myself I feel that once you start passing non-default parameters to `torch.nn.MSELoss(size_average=None, reduce=None, reduction='elementwise_mean')` it looses it allready lost its usefulness, i.e reduced code-complexity and readability.

**Comment by soumith** [comment]
we'll kind of leave it as-is for historical reasons, and also @ragulpr 's reasoning is sound.
